design-proposals/host-ip-pool.md

```diff
@@ -0,0 +1,442 @@
+# Host IP pooling with NNCPs.
+
+## Summary
+
+Provide a mechanism to assign static IP configuration for secondary interfaces
+for cluster nodes, supporting cluster scaling up and down.
+
+## Motivation
+
+Using DHCP as the IP assignment mechanism has some drawbacks:
+- Introduce latencies on node restart.
+- It presents a single point of failure in case the DHCP server is failing
+- It depends on network setup that can be complex.
+
+To overcome the above-mentioned drawbacks, statically configured interfaces taking the address from a
```

Comment:
The allocation and management of IP address on bare metal machines is a topic the IT industry had to tackle for the early days of computers.

I am sure DHCP and other options exists today which solve this challenge regardless of the kubernetes ecosystem.

I would suggest to examine existing enterprise and telco grade solutions that exist in the market, to make sure we are not re-inventing the wheel.

It is also not that trivial to claim that the DHCP drawbacks described above worth the investment of an in-house IP pool management. I would suggest first interviewing IT experts in this field before we commit to develop such a solution (which from first sight, looks very challenging and will require a lot of investment for the long run).